include/onnxruntime/core/framework/tensor_shape.h

```diff
@@ -135,7 +173,7 @@ class TensorShape {
   void Allocate(size_t size);
 
   gsl::span<int64_t> values_;
-  int64_t small_buffer_[5];
+  int64_t small_buffer_[kTensorShapeSmallBufferElementsSize];
```

Comment:
Do we still need to reinvent the implementation of inlined vectors given that we can now use abseil::inlined_vector?

Reply:
I had that thought. TensorShape is different. 1) it has to be able just to hold an external ptr for a special case. 2) It is immutable and so it is simpler.

Reply:
Ok, would be good to add a small note in case someone wonders and refactors this later.